crates/rust-analyzer/src/config.rs

```diff
@@ -1897,6 +1901,14 @@ impl Config {
         }
     }
 
+    fn cargo_all_targets(&self, source_root: Option<SourceRootId>) -> bool {
+        self.cargo_allTargets(source_root).unwrap_or_else(|| *self.cfg_setTest(source_root))
```

Comment:
Mainly because if you set `setTest` to false, you probably don't want `cargo check` to run on tests, and maybe not even build dev dependencies for them. `rust-analyzer.check.allTargets` defaults to `rust-analyzer.cargo.allTargets`, so I made it default to this.

I documented this:

> If you leave this unset but set `cfg.setTest` to `false`, this will also default to false.

Reply:
I find that a bit of an odd choice, `allTargets` includes more than just tests and if you don't want cargo check to check those well, you can just disable that config. check falling back to cargo configs is intentionally as those usually don't tend to diverge but this is an entirely different config.

Reply:
But if you don't want to set `cfg(test)`, you probably also don't want `cargo check` to do that, so what else can I do?

Reply:
Hmm judging from that issue that seems a separate matter to me. Either way imo if people already go out of their way to set that config they can also unset the all targets one